pkg/operator/controllers/checkers/clusterdnschecker/checker.go

```diff
@@ -14,7 +14,12 @@ import (
 )
 
 type clusterDNSChecker interface {
-	Check(ctx context.Context) error
+	Check(ctx context.Context) (result, error)
+}
+
+type result struct {
+	pass    bool
```

Comment:
Weak suggestion for `isSuccess` or just `success` instead of `pass`.

Reply:
done